tests/qat/test_quantum_backend.py

```diff
@@ -714,6 +714,163 @@ def test_create_duration_timeline_mapping(self):
             *[PostProcessing] * 4,
         ]
 
+    @pytest.mark.parametrize(
+        "hw", [get_test_model(), get_default_echo_hardware(), get_default_RTCS_hardware()]
+    )
+    def test_duration_timeline_times(self, hw):
+        # Tests that the creation of a duration timeline with a two-qubit circuit
+        # gives a timeline where each pulse is concurrent.
+        engine = hw.create_engine()
+        q1 = hw.get_qubit(0)
+        q2 = hw.get_qubit(1)
+
+        # Build a simple circuit
+        builder = (
+            get_builder(hw)
+            .X(q1, np.pi / 2.0)
+            .X(q2, np.pi)
+            .cnot(q1, q2)
+            .measure_mean_z(q1)
+            .measure_mean_z(q2)
+        )
+        res1 = engine.create_duration_timeline(builder.instructions)
+
+        # Check the times match up
+        for _, val in res1.items():
+            end = 0
+            for pos in val:
+                assert end == pos.start
+                end = pos.end
+
+    @pytest.mark.parametrize(
+        "hw", [get_test_model(), get_default_echo_hardware(), get_default_RTCS_hardware()]
+    )
+    def test_duration_timeline_sync(self, hw):
+        # Tests that a redundant sync has no effect on the circuit.
```

Comment:
makes sense as an ECR starts with a Sync. I would also try out another scenario where the added Sync would be on top of naturally/coincidentally synced timelines:

```
target1 : 1000 ns tune              + P
target2 : 500 ns tune + 500 ns tune + P
```

Everything that comes before P happens to align naturally, so adding a sync there would not practically affect timelines

Reply:
Very good point, addition test added!